tasks/read-a-configuration-file/src/main.rs

```diff
@@ -65,10 +65,8 @@ impl ConfigParams {
             }
         };
         let mut params = ConfigParams::new();
-        for line in content.lines().filter(is_not_comment) {
-            if let Ok(line) = line {
-                params.update_config(&line);
-            }
+        for line in content.lines().filter(is_not_comment).flatten() {
```

Comment:
Instead of flattening, I think a better strategy would be to bubble the errors up by modifying `parse` to return a `io::Result<ConfigParams>` . We could remove the `unwrap` opening the file as well.

Reply:
I made the changes, but I'm not sure I was able to follow your intentions, pls check.

Reply:
Yeah, this is closer. It can be improved further by making `is_not_comment` take a `&str` instead of a `&io::Result<String>`. That function shouldn't need to deal with the errors.

We may want to flip the return value and rename it `is_comment` to avoid a double negative, too.

```rust
for line in content.lines() {
    let line = line?;
    if is_comment(&line) {
        continue;
    }

    // ...
}
```

Reply:
Oh, we also might as well make `is_comment` a top-level function, no need for it to be a closure.